OpenSim/Sandbox/MatlabScripts/createActuatorsFile.m

```diff
@@ -0,0 +1,156 @@
+    if coordinate.isConstrained(state)
+        continue
+    end
+
+    % get the joint, parent and child names for the coordiante
+    joint = coordinate.getJoint();
+    parentName = joint.getParentFrame().getName();
+    childName = joint.getChildFrame().getName();
+
+    % If the coordinates parent body is connected to ground, we need to
+    % add residual actuators (torque or point).
+    if strmatch(parentName, model.getGround.getName() )
+
+        % if the joint type is custom or free
+        if strcmp(joint.getConcreteClassName(), 'CustomJoint') | strcmp(joint.getConcreteClassName(), 'FreeJoint')
+               % get the coordainte motion type
+               motion = char(coordinate.getMotionType);
+               % to get the axis value for the coordiante, we need to drill
+               % down into the coordinate transform axis
+               eval(['concreteJoint = ' char(joint.getConcreteClassName()) '.safeDownCast(joint);'])
+               sptr = concreteJoint.getSpatialTransform;
+               for ip = 0 : 5
+                  if strcmp(char(sptr.getCoordinateNames().get(ip)), char(coordinate.getName))
+                        sptr.getTransformAxis(ip).getAxis(axisValues);
+                        break
+                  end
+               end
+
+
+               % make a torque actuator if a rotational coordinate
+               if strcmp(motion, 'Rotational')
+                   newActuator = TorqueActuator(joint.getParentFrame(),...
+                                         joint.getParentFrame(),...
+                                         axisValues,...
+                                         1);
+
+               % make a point actuator if a translational coordainte.
+             elseif strcmp(motion, 'translational')
+                    % make a new Point actuator
+                    newActuator = PointActuator();
+                    % set the body
+                    newActuator.set_body(char(joint.getChildFrame().getName()))
+                    % set <point>      -0.07243760       0.00000000       0.00000000 </point>
+                    newActuator.set_point(massCenter)
+                    % set <point_is_global> false </point_is_global>
+                    newActuator.set_point_is_global(0)
+                    % set <direction>      -0.00000000       1.00000000      -0.00000000 </direction>
+                    newActuator.set_direction(axisValues)
+                    % set <force_is_global> true </force_is_global>
+                    newActuator.set_force_is_global(1)
+              else % something else that we don't support right now
+                    error(['Motion Type ' char(motion) 'not supported yet'])
+              end
+        else % if the joint type is not free or custom, just add coordinate actuators
+                % make a new coordinate actuator for that coordinate
+                newActuator = CoordinateActuator();
```

Comment:
`newActuator = CoordinateActuator();` creates a CoordinateActuator but doesn't specify which of the model's coordinates is being actuated. L146–153 set the optimal force (L147), the name of the actuator object (L149), and the min/max controls (L151–152), but not which coordinate to actuate.